src/components/TextInput.jsx

```diff
@@ -113,7 +114,7 @@ export default class TextInput extends Component {
     let value = event.target.value;
 
     if (!this.props.allowNewLine && value.match(newLineRegex) !== null) {
-      value = event.target.value.replace(newLineRegex, ' ');
+      value = value.replace(newLineRegex, ' ');
 
```

Comment:
I looked into it again today...
I think its not perfect implementation. 
1. In case of value/ valueLink - we do not need to update state with new value (without new lines), we can just call onUpdate with the new value and that should suffice.
2. We need to update state for defaultValue......but that makes me think might be it would be better to make textInput also a completed controlled component like comboBox.....
3. User 'has' here 'if (this.state.textareaProps.value)'

Also we should add a page in docs about how value/ defaultValue ad valueLink work with our onUpdate method.